packages/app-studio-toolkit/src/project-type/workspace-instance.ts

```diff
@@ -1,14 +1,38 @@
 import { WorkspaceImpl, WorkspaceApi } from "@sap/artifact-management";
 import * as vscode from "vscode";
+import { CommandExecutor } from "@sap/artifact-management";
+import { isPlainObject } from "lodash";
 
 let workspaceAPI: WorkspaceApi;
 
 /**
  * "Factory" for the `WorkspaceApi` "original" instance.
  */
-export function initWorkspaceAPI(): WorkspaceApi {
+export function initWorkspaceAPI(
+  context: vscode.ExtensionContext
+): WorkspaceApi {
   workspaceAPI = new WorkspaceImpl(vscode);
   workspaceAPI.startWatch();
 
+  context.subscriptions.push(
+    vscode.commands.registerCommand(
+      "project-api.command.run",
+      async (command, ...params) => {
+        try {
+          let result = await CommandExecutor.execute(command, ...params);
+          if (isPlainObject(result)) {
+            result = JSON.stringify(result, undefined, 2);
+          }
+          return !result ? "\n" : (result as string) + "\n";
```

Comment:
It seems strange that you add a `newline` to the results.
The formatting should be a concern of whoever prints those results out, not your concern...

Reply:
e.g maybe they use different newlines terminals?
or print the results in a table without newlines?

Reply:
also relying on the `!result` automatic boolean conversion to detect a string is not precise.

e.g: `!"" === true`

you are already using `isPlainObject` from lodash.
So you can also use isString.

If you want to add special handling for undefined
I would make this handling condition explicit:
`if x === undefined`

Otherwise this handler could also be run for other cases (e.g null)

Reply:
The formatting with newline still seems a little strange to me, but this is your flow and if you are sure
this is the optimal approach we can keep it.